Fix depends on when loading stack from file

teresaromero opened the pull request:
# Proposed changes

Fixes https://github.com/okteto/app/issues/6669

When reading a compose from a file using `okteto deploy -f docker-compose.yaml` the function that loads the stack was not performing the validation required.

This PR enables validation when the stack is read from file as it is done when inferring the stack when the command is `okteto deploy` and there is no `okteto.yaml` but `docker-compose.yaml`

Depends_on fields point into services within the same compose.

Output is now the same in both cases:

Example using `movies-with-compose` and removing the service `frontend` but leaving `depends_on` at `movies` service.

```
API server listening at: 127.0.0.1:16484
 i  Using teresaromero @ okteto.teresa.dev.okteto.net as context
 x   Service 'movies' depends on service 'frontend' which is undefined.
```

teresaromero commented on pkg/model/stack.go:
```diff
@@ -432,7 +432,7 @@ func (svc *Service) ToDev(svcName string) (*Dev, error) {
 	return d, nil
 }
 
-func (s *Stack) Validate() error {
```
it was just being used inside the pkg

teresaromero replied:
👍🏻  removed the commit

ifbyol commented on pkg/model/manifest.go:
```diff
@@ -410,12 +410,9 @@ func getManifestFromFile(cwd, manifestPath string) (*Manifest, error) {
 		}
-		s, stackErr := LoadStack("", composeFiles, false)
-
-		// We failed to load a stack file and a manifest file, so we need to return
-		// only the original manifest error
```
According to the code comment, this was done on purpose. Have we checked or asked about why we should be returning the original manifest error?

And why the code was getting a panic? My understanding is that this should be returning an error (not the one we wanted), but as it's an error, I would expect that the execution is broken instead of keep going and get a panic

ifbyol replied:
talked offline, we will run e2e tests to see if they pass or not with these changes

ifbyol replied:
@teresaromero should we also change it here? https://github.com/okteto/okteto/blob/bf5f3cde626c900e021f3b626c82a952bea2d77c/pkg/model/manifest.go#L440

I think that scenario is when we refer to a docker compose from an okteto manifest V2. If that docler-compose also have a depends_on, it might fail also as we are not validating the stack there.

teresaromero replied:
Yes, that is right, i've just debugged this and added validation. In this case there is also a panic if an okteto manifest has a compose section with an invalid compose...

In this case the err returned is this one

teresaromero commented:
> LGTM, it is possible to add a test for `getManifestFromFile` with an invalid stack?

I've refactored the tests for this function and added the case for the error on depends on

teresaromero commented:
Deepsource are already existent issues, i wont fix them as they imply changes not relative to this PR
